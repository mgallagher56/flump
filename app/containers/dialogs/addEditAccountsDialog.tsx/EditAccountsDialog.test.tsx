import type { ReactNode } from 'react';

import { fireEvent, render, waitFor } from '@testing-library/react';
import { vi } from 'vitest';

import mockUser from '__mocks__/user';

import AddEditAccountsDialogBtn from './AddEditAccountsDialog';

const mocks = vi.hoisted(() => ({
  mockUseLoaderData: vi.fn(),
  mockFrom: vi.fn(() => ({
    update: () => ({
      eq: () => ({
        eq: () => ({})
      })
    })
  }))
}));

vi.mock('@remix-run/react', async () => {
  const actual: Record<string, unknown> = await vi.importActual('@remix-run/react');
  return {
    ...actual,
    useLoaderData: mocks.mockUseLoaderData,
    Form: ({ children }: { children: ReactNode }) => <form>{children}</form>,
    useSubmit: () => ({ onSubmit: vi.fn() })
  };
});

vi.mock('app/utils/supabase', () => ({
  default: {
    from: mocks.mockFrom
  }
}));

describe('<EditAccountDialogBtn', () => {
  mocks.mockUseLoaderData.mockReturnValue({ user: mockUser });
  const { baseElement, getByText, getAllByText } = render(
    <AddEditAccountsDialogBtn accountId="123456" isEditAccount />
  );
  const triggerBtn = getByText('edit');
  expect(triggerBtn).toBeDefined();
  fireEvent.click(triggerBtn);
  test('should render edit account dialog when trigger button is clicked', async () => {
    await waitFor(() => {
      const addAccountBtn = getAllByText('save')[0];
      fireEvent.click(addAccountBtn);
      expect(mocks.mockFrom).toBeCalled();
    });
    expect(baseElement).toMatchSnapshot();
  });
});
