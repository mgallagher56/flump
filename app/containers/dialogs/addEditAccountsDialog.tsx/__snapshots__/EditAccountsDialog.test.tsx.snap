// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<EditAccountDialogBtn > should render edit account dialog when trigger button is clicked 1`] = `
<body
  style=""
>
  <div>
    <button
      aria-controls="dialog::r0::content"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="chakra-button css-4xx2wk"
      data-part="trigger"
      data-scope="dialog"
      data-state="closed"
      id="dialog::r0::trigger"
      type="button"
    >
      edit
    </button>
  </div>
</body>
`;
